### ISSUE TITLE
Not able to use Pattern on Ubuntu

### ISSUE THREAD

**Issue by termopro** [clips/pattern#76]
Hello,

I would like to report that Pattern couldn't be used out of the box with Ubuntu (12).

I have spent 4 hours today trying to figure out how to run python script which uses Pattern on Ubuntu and i've failed. The script uses SVM classification and the problems come from how Pattern supports it.

Problem N1
Pattern is dependent on 3rd party SVM classification library libsvm. It has to be installed on the system if you wish to use SVM in pattern, otherwise Pattern dies with exception.

Problem N2
Even after you install libsvm by hand using "sudo apt-get libsvm3" you will not be able to use it. 

You will get the following error:
"AttributeError: /usr/lib/libsvm.so.3: undefined symbol: svm_get_sv_indices"

To repeat this problem, simply try to run this example on Ubuntu
(https://github.com/clips/pattern/blob/master/examples/01-web/08-wiktionary.py)

The problem seems to happen because Pattern requires newer libsvm module than the one being installed with "apt-get".

What am i doing wrong ?
How do i properly install Pattern along with libsvm on Ubuntu ?

**Comment by priyolahiri** [comment]
@termopro did you manage to fix it? if yes, how?

**Comment by qacollective** [comment]
In the hope I might save someone some time here ... I found this issue when searching for the exact same error for another reason.

The only way I was able to resolve it was my downloading libsvm and compiling it to produce an additional library file that doesn't seem to come in any of the ubuntu packages.

See the following URL for the commands I used to solve the problem for SideKit - the line which copies the resultant library (libsvm.so.2) would probably change for your project and may just need to go in /usr/lib.

I'm aware that your error mentions libsvm.so.3 as opposed to libsvm.so.2, but I had the exact same error and it seems svm_get_sv_indicies only appears in libsvm.so.2.

Andrew
